### PR TITLE
Added configurable Google Tag Manager domain.

### PR DESCRIPTION
**Pull request by alinistrate**
### Ticket
- [Make GTM script domain configurable in plugin settings](https://app.asana.com/0/1201940451841784/1202508895538464)

### CODE REVIEW

**Review by bogdanarizancu — APPROVED** [review]

File: src/Plugin.php [hunk 1]
```diff
@@ -87,8 +87,9 @@ public static function wp_head() {
     if (!$gtm_id = get_option('shop_analytics_gtm_id')) {
       return;
     }
+    $embed_url_domain = get_option('shop_analytics_url_domain') ?: 'www.googletagmanager.com';
```
Comment (sun) [inline src/Plugin.php:90]: See https://github.com/netzstrategen/wordpress-cookie-consent/pull/25#discussion_r923313235
```suggestion
    $embed_url_domain = get_option('shop_analytics_gtm_domain') ?: 'www.googletagmanager.com';
```

File: templates/settings.php [hunk 2]
```diff
@@ -201,6 +201,23 @@
...
+          <p class="hint"><?= __('Default: www.googletagmanager.com', Plugin::L10N) ?></p>
+        </th>
+        <td>
+          <input
+            id="shop-analytics-url-domain"
+            name="shop_analytics_url_domain"
+            type="text"
+            size="50"
+            value="<?= esc_attr(get_option('shop_analytics_url_domain')) ?>"
+            maxlength="40"
```
Comment (sun) [inline templates/settings.php:216]:
```suggestion
```

File: templates/settings.php [hunk 3]
```diff
@@ -201,6 +201,23 @@
           </td>
         </tr>
       <?php endif; ?>
+      <tr class="form-field">
+        <th scope="row">
+          <label for="shop-analytics-url-domain"><?= __('Script embed URL domain', Plugin::L10N) ?></label>
```
Comment (sun) [inline templates/settings.php:206]:
```suggestion
          <label for="shop-analytics-url-domain">Google Tag Manager Domain</label>
```

**Review by sun — CHANGES_REQUESTED** [review]

File: src/Admin.php [hunk 4]
```diff
@@ -37,6 +37,7 @@ public static function init() {
     register_setting('shop-analytics-settings', 'shop_analytics_brand_attribute');
     register_setting('shop-analytics-settings', 'shop_analytics_product_id');
     register_setting('shop-analytics-settings', 'shop_analytics_category_attribute');
+    register_setting('shop-analytics-settings', 'shop_analytics_url_domain');
```
Comment (sun) [inline src/Admin.php:40]:
```suggestion
    register_setting('shop-analytics-settings', 'shop_analytics_gtm_domain');
```

File: src/Plugin.php [hunk 5]
```diff
@@ -99,8 +100,9 @@ public static function wp_footer() {
...
```
Comment (sun) [inline src/Plugin.php:103]:
```suggestion
    $embed_url_domain = get_option('shop_analytics_gtm_domain') ?: 'www.googletagmanager.com';
```

File: templates/settings.php [hunk 6]
```diff
@@ -201,6 +201,22 @@
           </td>
         </tr>
       <?php endif; ?>
+      <tr class="form-field">
+        <th scope="row">
+          <label for="shop-analytics-url-domain">Google Tag Manager Domain</label>
```
Comment (sun) [inline templates/settings.php:206]:
```suggestion
          <label for="shop-analytics-gtm-domain">Google Tag Manager Domain</label>
```

File: templates/settings.php [hunk 7]
```diff
@@ -201,6 +201,22 @@
...
+          <p class="hint"><?= __('Default: www.googletagmanager.com', Plugin::L10N) ?></p>
+        </th>
+        <td>
+          <input
+            id="shop-analytics-url-domain"
+            name="shop_analytics_url_domain"
+            type="text"
+            size="50"
+            value="<?= esc_attr(get_option('shop_analytics_url_domain')) ?>"
```
Comment (sun) [inline templates/settings.php:215]:
```suggestion
            id="shop-analytics-gtm-domain"
            name="shop_analytics_gtm_domain"
            type="text"
            size="50"
            value="<?= esc_attr(get_option('shop_analytics_gtm_domain')) ?>"
```

**Review by sun — CHANGES_REQUESTED** [review]
The other instances of the names were not updated accordingly.

**Review by sun — APPROVED** [review]
Thank you! 👍  Please squash-merge using the PR title as commit message.

**Pull request merged**